src/game/client/input/keyboard.cpp

```diff
@@ -134,17 +134,27 @@ void Keyboard::Create_Stream_Messages()
     if (g_theMessageStream != nullptr) {
         GameMessage *msg = nullptr;
 
-        for (int i = 0; m_keys[i].key != 0; ++i) {
-            if (m_keys[i].state & KEY_DOWN) {
+        KeyboardIO *ptr = Get_First_Key();
+        while (ptr->key != 0) {
```

Comment:
I agree with @duncanspumpkin on this one, it would be much better as a for loop.
`for (KeyboarIO *keyio = Get_First_Key; keyio->key != 0; ++keyio)`